ui/src/pages/trading/Market.tsx

```diff
@@ -245,22 +255,38 @@ export const Market: React.FC<ServicePageProps<Service> & Props> = ({
         timeInForce === 'GTD'
           ? { tag: 'GTD', value: { expiryDate: expiryDate.toString() } }
           : { tag: timeInForce, value: {} },
+      optExchangeFee: null,
     };
     clearOrderForm();
     if (listing.payload.listingType.tag === 'Collateralized') {
       const depositCid = isBuy
         ? await getAsset(quotedAssets, price * quantity)
         : await getAsset(tradedAssets, quantity);
-
       if (!depositCid) return;
+      let optExchangeFee = null;
+      if (feeAmount > 0) {
+        const feeAssets = await ledger
+          .query(AssetDeposit)
```

Comment:
should we gather assetDeposits independently with `useStreamQueries`? I hvnt seen `.query` used in this app yet, but your call.

Reply:
The issue here is that the asset deposits from the `useStreamQuery` may be changed in the preceding `getAsset` (if the currencies were the same) and the stream wasn't getting updated quickly enough before calling `getAsset` again here, performing a manual query fixed this.